### ISSUE TITLE
Create unit tests for the memory cache

### ISSUE THREAD

**Issue by pbrezina** [pbrezina/sssd-test#2059]
Cloned from Pagure issue: https://pagure.io/SSSD/sssd/issue/2055
- Created at 2013-08-24 18:33:02 by [jhrozek](https://pagure.io/user/jhrozek)
- Closed at 2020-03-24 14:18:18 as wontfix
- Assigned to [lslebodn](https://pagure.io/user/lslebodn)
---

The memcache has seen quite a few patches lately and in general is a critical component, so we should add unit tests. Lukas came up with helper functions to exercise the memcache, which could be used as a starting point

### Comments

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2013-08-27 10:58:24](https://pagure.io/SSSD/sssd/issue/2055#233238)**

Patched are attached in mail:
https://lists.fedorahosted.org/pipermail/sssd-devel/2013-August/016163.html

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2013-08-29 15:52:24](https://pagure.io/SSSD/sssd/issue/2055#233239)**

Fields changed

milestone: NEEDS_TRIAGE => SSSD 1.12 beta
rhbz:  => 0

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2014-02-19 11:55:27](https://pagure.io/SSSD/sssd/issue/2055#233240)**

Fields changed

blocking:  => #1923
owner: somebody => lslebodn

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2014-05-30 15:54:00](https://pagure.io/SSSD/sssd/issue/2055#233241)**

Fields changed

milestone: SSSD 1.12 beta => SSSD 1.12.1

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2014-09-08 20:09:07](https://pagure.io/SSSD/sssd/issue/2055#233242)**

Mass-moving all tickets that didn't make 1.12.1 into 1.12.2

milestone: SSSD 1.12.1 => SSSD 1.12.2

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2014-09-30 19:06:23](https://pagure.io/SSSD/sssd/issue/2055#233243)**

We need to do a release as requested by downstream. Moving tickets that are not fixed already or very close to acking to 1.12.3

milestone: SSSD 1.12.2 => SSSD 1.12.3

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2014-11-19 17:37:54](https://pagure.io/SSSD/sssd/issue/2055#233244)**

It makes sense to write the unit tests now instead of moving them to a later release again, but the tickets in 1.12.3 should be more focused at this point. Moving to 1.12.4

mark:  => 0
milestone: SSSD 1.12.3 => SSSD 1.12.4

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2015-02-12 20:05:41](https://pagure.io/SSSD/sssd/issue/2055#233245)**

Fields changed

milestone: SSSD 1.12.4 => SSSD Deferred

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2016-11-17 11:40:45](https://pagure.io/SSSD/sssd/issue/2055#233246)**

I suggest we close this ticket because we now have integration tests for memory cache.

review: 0 => 1
sensitive:  => 0

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2016-11-25 09:48:55](https://pagure.io/SSSD/sssd/issue/2055#233247)**

Fields changed

resolution:  => invalid
status: new => closed

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2016-11-25 10:00:22](https://pagure.io/SSSD/sssd/issue/2055#233248)**

Integration test cannot cover all weird corner cases.

resolution: invalid => 
status: closed => reopened

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2017-02-24 14:47:18](https://pagure.io/SSSD/sssd/issue/2055#233249)**

**Metadata Update from @jhrozek**:
- Issue assigned to lslebodn
- Issue marked as blocked by: #1923
- Issue set to the milestone: SSSD Patches welcome

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2020-03-24 14:18:16](https://pagure.io/SSSD/sssd/issue/2055#636886)**

Thank you for taking time to submit this request for SSSD. Unfortunately this issue was not given priority and the team lacks the capacity to work on it at this time.

Given that we are unable to fulfill this request I am closing the issue as wontfix.

If the issue still persist on recent SSSD you can request re-consideration of this decision by reopening this issue. Please provide additional technical details about its importance to you.

Thank you for understanding.

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2020-03-24 14:18:18](https://pagure.io/SSSD/sssd/issue/2055#636887)**

**Metadata Update from @pbrezina**:
- Issue close_status updated to: wontfix
- Issue status updated to: Closed (was: Open)
